main.js

```diff
@@ -14,9 +14,100 @@
 // limitations under the License.
 //------------------------------------------------------------------------------
 
-let alexaVerifier = require('alexa-verifier');
-let ConversationV1 = require('watson-developer-cloud/conversation/v1');
-let redis = require("redis");
+const alexaVerifier = require('alexa-verifier');
+const ConversationV1 = require('watson-developer-cloud/conversation/v1');
+const redis = require("redis");
+const openwhisk = require('openwhisk');
+const request = require('request');
+
+function lookupGeocode(args, location) {
+
+    if (args.WEATHER_URL) {
+        return new Promise(function(resolve, reject) {
+            const weatherPort = args.WEATHER_PORT || 443;
+            const url = args.WEATHER_URL + ':' + weatherPort + '/api/weather/v3/location/search';
+            console.log("Getting geocode for " + location);
+            request({
+                method: 'GET',
+                url: url,
+                jar: true,
+                json: true,
+                qs: {
+                    query: location,
+                    locationType: 'city',
+                    language: 'en-US'
+                }
+            }, function (err, response, body) {
+                // console.log("Locations from Weather location services");
+                // console.log(body.location);
+                if (body.location.length < 1) {
+                    reject("Location not found");
+                }
+                // Just take the first one.
+                let latitude = body.location.latitude[0];
+                let longitude = body.location.longitude[0];
+                resolve([latitude, longitude]);
+            });
+        });
+    }
+    else {
+        console.log("Cannot lookup geocode, using Honolulu.");
```

Comment:
good choice

Reply:
I forgot to mention error handling needs work. Plan to address that when test coverage is added.  Might use Honolulu when geocode doesn't work, but for general error cases I was thinking maybe just say "Winter is coming"